packages/form-js-editor/src/render/components/FieldResizer.js

```diff
@@ -0,0 +1,184 @@
+
+const GRID_OFFSET_PX = 16;
+
+
+export function FieldResizer(props) {
+
+  const {
+    field,
+    position
+  } = props;
+
+  const ref = useRef(null);
+
+  const formLayoutValidator = useService('formLayoutValidator');
+  const modeling = useService('modeling');
+
+  // we can't use state as we need to
+  // manipulate this inside dragging events
+  const context = useRef({
+    startColumns: null,
+    newColumns: null
+  });
+
+  // create a blank canvas to use as drag preview
+  // ensure it was only created once
+  useEffect(() => {
+    let blankCanvas = getDragPreviewImage();
+
+    if (!blankCanvas) {
+      blankCanvas = document.createElement('canvas');
+      blankCanvas.classList.add(RESIZE_DRAG_PREVIEW_CLS);
+      document.body.appendChild(blankCanvas);
+    }
+  }, []);
+
+  const onResize = throttle((_, delta) => {
+    const { x: dx } = delta;
+
+    const { layout = {} } = field;
+
+    const newColumns = calculateNewColumns(
+      ref.current,
+      layout.columns || context.current.startColumns,
+      dx,
+      position
+    );
+
+    const errorMessage = formLayoutValidator.validateField(field, newColumns);
+
+    if (!errorMessage) {
+
+      context.current.newColumns = newColumns;
+
+      // make visual updates to preview change
+      const columnNode = ref.current.closest('.fjs-layout-column');
+      removeMatching(columnNode, COLUMNS_REGEX);
+      columnNode.classList.add(`cds--col-lg-${newColumns}`);
+    }
+  });
+
+  const onResizeStart = (event) => {
+    const onDragStart = createDragger(onResize, getDragPreviewImage());
+    onDragStart(event);
+
+    const target = getElementNode(field);
+
+    // mitigate auto columns on the grid that
+    // has a offset of 16px (1rem) to both side
+    const columnNode = getColumnNode(target);
+    const startWidth = columnNode.getBoundingClientRect().width + GRID_OFFSET_PX;
+    context.current.startColumns = asColumns(startWidth, getParent(target));
+
+    setResizing(target, position);
+  };
+
+  const onResizeEnd = () => {
+    const { layout = {} } = field;
+
+    if (context.current.newColumns) {
+      modeling.editFormField(field, 'layout', {
+        ...layout,
+        columns: parseInt(context.current.newColumns)
+      });
+    }
+
+    const target = getElementNode(field);
+    unsetResizing(target, position);
+
+    context.current.newColumns = null;
+  };
+
+  if (field.type === 'default') {
+    return null;
+  }
+
+  return (
+    <div
+      ref={ ref }
+      class={ classNames(
+        'fjs-field-resize-handle',
+        'fjs-field-resize-handle-' + position,
+        DRAG_NO_MOVE_CLS
+      ) }
+      draggable={ true }
+      onDragStart={ onResizeStart }
+      onDragEnd={ onResizeEnd }>
+    </div>
+  );
+}
+
+
+// helper //////
+
+function asColumns(width, parent) {
+  const totalWidth = parent.getBoundingClientRect().width;
+
+  const oneColumn = (1 / 16) * totalWidth;
+
+  return Math.round(width / oneColumn);
```

Comment:
That's true, but we anyway [permit uneven columns via the `FormLayoutValidator`](https://github.com/bpmn-io/form-js/pull/593/files#diff-2844e6cf69aed3c32393e058a5cf19ff3b2414c1961caa1738fb5eace10ff57aR29), that's why a check for even numbers is not needed here. I wanted to keep the helper less restrictive, so I left it out here. 

In the end, we should have the expected behavior of even grid steps.

Reply:
I know we don't permit uneven columns, but this isn't the problem I'm referring to. 

If you look carefully in the below example, you can see that the rounding causes the dragging to not be 'centered', it starts a little bit to the left (1/4 of 2 rows), and ends quite a bit to the right (3/4 of 2 rows). This is because the maths doesn't work out here because you're essentially still rounding single columns but using that result to calculate positions in a 2 column system. 

![firefox_oFwPqYezIc](https://user-images.githubusercontent.com/17801113/232463685-6b21fae0-03b2-4b7e-83ad-7c53488a2148.gif)

Reply:
![image](https://user-images.githubusercontent.com/17801113/232465116-b7fbfeb3-e533-446e-b599-dfdbc46ff2eb.png)